mmaction/models/backbones/resnet3d.py

```diff
@@ -813,3 +815,249 @@ def train(self, mode=True):
             for m in self.modules():
                 if isinstance(m, _BatchNorm):
                     m.eval()
+
+
+@BACKBONES.register_module()
+class ResNet3dLayer(nn.Module):
```

Comment:
Is it possible to inherit from ResNet3d? It seems that most codes are duplicated

Reply:
We can do that, but I think making ResNet3dLayer inherit from ResNet3d does not make sense. Since it doesn't have some attributes or functions that ResNet3d has.

Reply:
Which attributes and why they cannot be solved

Reply:
Or try to reduce code duplication?

![image](https://user-images.githubusercontent.com/9464825/100704567-af335580-33e0-11eb-8683-4c6ad238a4be.png)

Reply:
For example, ResNet3dLayer:
will not have attributes like `conv1_kernel`, `conv1_stride_t`, `frozen_stages`(since only 1 stage in Layer), `with_pool2`, etc., 
will not have methods like `make_stem_layer`
functions including `__init__`, `_freeze_stages`, `forward` will be overriden
and ResNet3dLayer is not a ResNet3d (on the contrary, ResNet3dSlowOnly is ResNet3d, etc.)

Although we don't inherit from ResNet3d now, the only duplicated code is `init_weights` and `_inflate_weights`, maybe that is acceptable?

Reply:
![image](https://user-images.githubusercontent.com/9464825/100705492-3a611b00-33e2-11eb-8ad1-cfc7a98229da.png)

Reply:
That's a good idea, I will update my code with this style if it is permitted.

Reply:
You can change the name `self` to avoid confusion

Reply:
Although I think inheritance is the better solution. You can jump over the init of ResNet3d and call directly call the init of Module, this way no extra variables will be initialized